### DISCUSSION TITLE
Wiegand CoasterKart

### DISCUSSION THREAD

**Discussion by spacek531** [OpenRCT2/OpenRCT2#20938] [Feature requests]
It was revealed that Wiegand, along side with approving the use of their alpine coaster model, had mentioned they would like to see their CoasterKart model in the game. This discussion is my announcement of working on this feature.

## Overview

https://www.wiegandslide.com/en/products/coasterkart.html

https://www.youtube.com/watch?v=x4mRJxdjMPQ

The Wiegand CoasterKart can travel at a top speed of 40 km/h. The vehicles move up autonomously in the station when a vehicle is dispatched from the loading platform. The vehicles seat 2 people and are stylized to look like an off-road vehicle. The riders can press the pedal to go faster or slower. The ride layouts feature lots of shallow banked turns and sloped banked turns with lots of landscaping, and the tracks do not go very high above the ground, approximately 4 meters max.

Current questions:
- Do the vehicles require at least one rider to operate?
- Do the vehicles have a minimum speed on the main circuit?
- ~~Do the rides have enforced speed zones?~~ yes

## Implementation

The implementation will be relatively straightforward. The ride will be set up like a simple tracked ride that uses powered cars. The ride will allow dispatching of empty vehicles in test mode. The ride will have all vehicles move with standard powered car behavior and will not feature speed variances based on the guest. I may add the ability to set the car's speed based on where it is on the track. Here is a mockup using a mine ride:

![image](https://github.com/OpenRCT2/OpenRCT2/assets/3454156/b76808e6-dfef-4c78-b399-de34db81c0b9)

The ride vehicle is straightforward. It will have a powered speed set no higher than 40 km/h and it will have animated restraints. Riders will ride in pairs.

The ride tracks will be made custom to match the real ride using @X123M3-256's track generator tool. The track elements will include sloped turns, banked sloped turns, and the diagonal variants of these. I intend to use this project to investigate 22.5 degree bankings for these elements. I also intend to use this project to develop ways of objectifying track types by developing structures and methods to streamline paint code.

**Comment by ZehMatt** [comment]
I think we should also have this the last one to be done the traditional way in the code and move forward towards objects after that, we need to de-hardcode this to make it easier in the future to allow being rides/tracks added a lot easier.

**Comment by spacek531** [comment]
Opinions on if the ride should be classified as coaster or thrill ride?

**Reply by astraylife1** [reply]
Seems more of a thrill ride when looking at similar-ish rides like EPCOT's test track or RCT's monorail cycles.

**Reply by karst** [reply]
I am more for it to be a thrill ride. I can see it being classified as a roller coaster though, because the powered coaster is, but it doesn't coast, so there's that...

**Reply by karst** [reply]
We can also ask RCDB, does that classify it as a coaster?

**Reply by spacek531** [reply]
I've seen enough arguments here and on discord to decide that it will be a thrill ride.

**Comment by spacek531** [comment]
Opinions on the ingame name and description of the ride?

Currently I have:
`Coaster Karts`
`Linear induction motors propel vehicles along the track, with riders controlling the speed using a foot pedal`

The description is not super great because I don't plan to make the speed change with riders like the alpine coaster, but I am planning to allow the user to set the speed on each track element so the cart can go different speeds at specific parts.

**Reply by karst** [reply]
Is it not possible to reuse some of the alpine coaster seeds where you can set a speed limit but the peeps can still decide the speed until that point? Or are you planning that the CK has a test mode too (in which case it can just use full speed?)

**Comment by spacek531** [comment]
Opinions on how high the track should be? Line up with giga/mini? Others? Don't care?

Even if it is the same height as giga/mini the rails will not line up perfectly

**Reply by karst** [reply]
I think lining it up as much as possible with the alpine coaster would be best, The track is pretty similar, though prolly a lil wider.

**Comment by spacek531** [comment]
Opinions on how many vehicle types should be included?

The minimum is one, obviously. It's this one:

![160113993_930408181133869_7060664794088760402_o](https://github.com/OpenRCT2/OpenRCT2/assets/3454156/b27e8ad6-fdfb-4f60-b532-e04466765983)

Although Wiegand has not advertised vehicle theming that substantially changes the panels, I think OpenRCT2 could add at least one alternative design. Potential themes include:
- Full-body sports car (Jaguar E-Type, Aston Martin DB-3, Nissan 240Z, Fiat 1200)
- 'Winged' sports car (MG midget, Packard 115-C)
- Off-road vehicle (Suzuki Samurai, Willies Jeep, Jeep Wrangler, Land Rover Defender)
- 18-wheeler tractor (Peterbilt 359, Freightliner FLC)
- Steam locomotive (Metropolitan Railway # 23, Talyllyn Railway 'Dolgoch')
- Extraterrestrial Rover

**Reply by karst** [reply]
Extra theming seems like a really cool idea.

**Reply by Manticore-007** [reply]
I like the sound of the extraterrestrial rover!

**Reply by deurklink** [reply]
The extraterrestrial rover could be great for a space / martian theme! The off-road vehicle would be great for a jungle / desert theme.

**Reply by ocalhoun6** [reply]
Eh, I'd say start it with just the one vehicle type, and others could be added after the rides/vehicles as objects system is implemented. Maybe a second vehicle for this ride could be one of the first that's added as an object.

**Reply by karst** [reply]
The vehicles are luckily already objects so that wouldn't be a botlteneck here.

**Reply by spacek531** [reply]
> Eh, I'd say start it with just the one vehicle type, and others could be added after the rides/vehicles as objects system is implemented. Maybe a second vehicle for this ride could be one of the first that's added as an object.

ZehMatt was referring to ride type objects and track type objects. As karst said, vehicle objects are already a thing and have been since 2002, so there is nothing stopping another one from being added other than the dev's vision.

**Reply by spacek531** [reply]
To my Extraterrestrial Rover lovers, please find or create some references for a rover that can be transplanted onto the chassis of a CoasterKart. Extra points if it fits the theme of the existing Martian scenery 
![image](https://github.com/OpenRCT2/OpenRCT2/assets/3454156/a425557a-22c3-4419-8e3a-48bc405498fc)
